pkg/runtime/dispatcher.go

```diff
@@ -0,0 +1,443 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package runtime
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"sync/atomic"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/proto"
+	rpc "github.com/googleapis/googleapis/google/rpc"
+	"github.com/hashicorp/go-multierror"
+	"github.com/opentracing/opentracing-go"
+	tracelog "github.com/opentracing/opentracing-go/log"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"istio.io/mixer/pkg/adapter"
+	adptTmpl "istio.io/mixer/pkg/adapter/template"
+	"istio.io/mixer/pkg/aspect"
+	"istio.io/mixer/pkg/attribute"
+	cpb "istio.io/mixer/pkg/config/proto"
+	"istio.io/mixer/pkg/expr"
+	"istio.io/mixer/pkg/pool"
+	"istio.io/mixer/pkg/status"
+	"istio.io/mixer/pkg/template"
+)
+
+// Action is the runtime representation of configured action - cpb.Action.
+// Configuration is processed to hydrate instance names to Instances and handler.
+type Action struct {
+	// ready to use handler
+	handler adapter.Handler
+	// configuration for the handler
+	handlerConfig *cpb.Handler
```

Comment:
Ah, I see below it's fetching some strings out of this config. Then let's make Action just hold those strings instead of the whole config. Otherwise, it really feels like a layering problem.